src/rulesets/LinearAlgebra/norm.jl

```diff
@@ -111,8 +108,9 @@ end
 
 function _normp_back_x(x, p, y, Δy)
     c = real(Δy) / y
-    ∂x = similar(x)
-    broadcast!(∂x, x) do xi
+    T = promote_type(eltype(x), typeof(c))
+    ∂x = similar(x, T)  # same comment as _norm1_back about allocation and type-stability.
+    ∂x = broadcast!(∂x, x) do xi
```

Comment:
Probably doesn't really matter much, but I found broadcast to sometimes be a bit tricky w.r.t. type inference, especially when working with `CuArrays`. Since `∂x` and `x` have the same dimensions anyways, I would probably use `map!` just to be safe:
```suggestion
    map!(∂x, x) do xi
```

Reply:
Sure, it should be simpler. (For tuples I've certainly seen it behave better.)